client/src/data/ide-presets.yaml

```diff
@@ -0,0 +1,74 @@
+vscode:
```

Comment:
Is it really better to add all this stuff (webpack plugin, .d.ts file) than simply making this a .ts file? 🤔

Reply:
It definitely over the top but we already rely on the `yaml` npm package anyway (we even ship it as runtime dep) and type definitions would be necessary in both cases. I think the yaml file improves readability but of course adds some complexity. Is it worth it?

Reply:
Hmm good point. The type definition is probably not necessary because it can be inferred 🤔 Idk, we make use of YAML a lot in the project already, so it's probably fine.